File: 2037-count-square-sum-triples/2037-count-square-sum-triples.java
class Solution {
    public int countTriples(int n) {
        int count=0;
        for (int i = 1; i <= n; i++) {
            for (int j = i + 1; j <= n; j++) {
                int sumSquares = (i * i) + (j * j);
                int c = (int) Math.sqrt(sumSquares);
                if (c <= n && c * c == sumSquares) {
                    count++;
                }
            }
        }
        return count*2;
    }
}

File: 2037-count-square-sum-triples/2037-count-square-sum-triples.rs
impl Solution {
    pub fn count_triples(n: i32) -> i32 {
        let mut count =0;
        for i in 1..=n {
            for j in i+1..=n{
                let sum_square = (i*i) + (j*j);
                let mut c = (sum_square as f64).sqrt() as i32;

                if(c<=n && c*c==sum_square){
                    count =count+1;
                }
            }
        }
        count*2
    }
}